pkg/plugin/backup_pvc_action_plugin.go

```diff
@@ -107,7 +107,7 @@ func (p *NewPVCBackupItemAction) Execute(item runtime.Unstructured, backup *vele
 
 	p.Log.Info("Creating a Snapshot CR")
 	updatedSnapshot, err := snapshotUtils.SnapshotRef(ctx, backupdriverClient, objectToSnapshot, pvc.Namespace, *backupRepository,
-		[]backupdriverv1.SnapshotPhase{backupdriverv1.SnapshotPhaseSnapshotted, backupdriverv1.SnapshotPhaseSnapshotFailed}, p.Log)
+		[]backupdriverv1.SnapshotPhase{backupdriverv1.SnapshotPhaseSnapshotted, backupdriverv1.SnapshotPhaseUploaded, backupdriverv1.SnapshotPhaseSnapshotFailed}, p.Log)
```

Comment:
The waitForPhases is an OR of all the status passed, so this will not really wait for Uploaded phase and return as soon as Snapshotted is reached. Furthermore, in GC it will never reach Uploaded phase.

Reply:
Snapshotted is not used right now actually.  I think we probably should set Snapshotted phase for local mode?

On GC, the snapshot status should be updated based Supervisor Cluster so we should be waiting for the same statuses.

Reply:
I think we should also wait for SnapshotPhaseCanceled.
What about SnapshotPhaseCleanupFailed?  Is that a terminal state?

Reply:
Removed "Snapshotted" for now as we are not using it.  We can add it back if it is needed in the future.
Added SnapshotPhaseCanceled.

Reply:
From my observations on GC setup, the Snapshot status eventually moves to "Uploaded" on the supervisor. The Snapshot status on Guest remains at "Snapshotted".
SnapshotPhaseCanceled is not yet supported, so no point.
Re-iterating it again, the waitForPhases is an OR of all the status passed, so it will exit on the earliest state in the list, so SnapshotPhaseCleanupFailed wouldn't make a difference.

@swgupta-vmw can chime in.

Reply:
Seems like thats the GC not updating to "Uploaded" is a fall out from my change. Uploaded/Failed should suffice.

Reply:
Discussed with Lintong.  We still should wait for Snapshotted in the plugin as plugin will not wait for upload completion.

Reply:
I think exiting on the OR is fine.  There's no guarantee we will see all intermediate states, right?

Reply:
Yeah, there could be race conditions.

Modified code to wait for all statuses except for New and InProgress.
For SnapshotPhaseCleanupFailed, since upload is already complete so I logged an error message but didn't return error.  This way the backup will still be marked as successful.